src/openfermion/measurements/_rdm_mapping_functions.py

```diff
@@ -0,0 +1,200 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Mapping RDMs to other RDMs"""
+import numpy
+from itertools import product
+
+
+def krond(i, j):
+    return 1 if i == j else 0
+
+
+def tpdm_to_opdm(tpdm, particle_number):
+    """
+    Contract a 2-RDM to a 1-RDM
+
+    Args:
+        tpdm (numpy.ndarray): The 2-RDM as a 4-index tensor. Indices follow the
+            internal convention of tpdm[p, q, r, s] ==
+            :math:`a_{p}^{\dagger}a_{q}^{\dagger}a_{r}a_{s}`.
+        particle_number (float): number of particles in the system
+
+    Returns:
+        opdm (numpy.ndarray): The 1-RDM contracted from the tpdm.
+    """
+    return numpy.einsum('prrq', tpdm) / (particle_number - 1)
```

Comment:
I don't think we can reasonably expect `particle_number` to be a float, cast `(particle_number - 1)` explicitly

Reply:
Indeed, in many cases the user will provide an int which would make this function perform int division.

Reply:
I think the best way to deal with division is to add `from __future__ import division` at the top of the file. Then we can just use Python 3 syntax.

Reply:
👍